sdc/Sdc.i

```diff
@@ -1270,30 +1270,41 @@ all_outputs_cmd()
   return sta->sdc()->allOutputs();
 }
 
-PortSeq
-filter_ports(const char *property,
-	     const char *op,
-	     const char *pattern,
-	     PortSeq *ports)
-{
-  PortSeq filtered_ports;
-  if (ports) {
+template <typename T> Vector<T*>
+filter_objects(const char *property,
+         const char *op,
+         const char *pattern,
+         Vector<T*> *objects)
+{
+  Vector<T*> filtered_objects;
+  if (objects) {
     Sta *sta = Sta::sta();
     bool exact_match = stringEq(op, "==");
     bool pattern_match = stringEq(op, "=~");
     bool not_match = stringEq(op, "!=");
-    for (const Port *port : *ports) {
-      PropertyValue value(getProperty(port, property, sta));
+    bool not_pattern_match = stringEq(op, "!~");
+    for (T *object : *objects) {
+      PropertyValue value(getProperty(object, property, sta));
       const char *prop = value.stringValue();
```

Comment:
this has to be asString; stringValue only applies to properties that have string value (stringValue() should probably error to check)

Reply:
Fixed